js/flexbuffers.js

```diff
@@ -0,0 +1,1015 @@
+var flexbuffers = {};
+
+flexbuffers.BitWidth = {
+  WIDTH8: 0,
+  WIDTH16: 1,
+  WIDTH32: 2,
+  WIDTH64: 3,
+};
+
+flexbuffers.BitWidth.toByteWidth = (bitWidth) => {
+  return 1 << bitWidth;
+};
+
+flexbuffers.BitWidth.width = (value) => {
+  if (Number.isInteger(value)) {
+    const v = value < 0 ? value * -1 : value;
+    if (v >> 7 === 0) return flexbuffers.BitWidth.WIDTH8;
+    if (v >> 15 === 0) return flexbuffers.BitWidth.WIDTH16;
+    if (v >> 31 === 0 && v >> 32 === v) return flexbuffers.BitWidth.WIDTH32;
+    return flexbuffers.BitWidth.WIDTH64;
+  }
+
+  function _toF32(value) {
+    const buffer = new ArrayBuffer(4);
+    const view = new DataView(buffer);
+    view.setFloat32(0, value, true);
+    return view.getFloat32(0, true);
+  }
+
+  return value === _toF32(value) ? flexbuffers.BitWidth.WIDTH32: flexbuffers.BitWidth.WIDTH64;
+};
+
+flexbuffers.BitWidth.fwidth = (value) => {
+  function _toF32(value) {
+    const buffer = new ArrayBuffer(4);
+    const view = new DataView(buffer);
+    view.setFloat32(0, value, true);
+    return view.getFloat32(0, true);
+  }
+
+  return value === _toF32(value) ? flexbuffers.BitWidth.WIDTH32: flexbuffers.BitWidth.WIDTH64;
+};
+
+flexbuffers.BitWidth.uwidth = (value) => {
+  if (Number.isInteger(value) && value >= 0) {
```

Comment:
we're checking types for inputs in multiple places, wouldn't it be better to make separate methods for ints and floats so we don't need these checkes?

Reply:
I removed all defensive checks